v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -68,17 +68,19 @@ func (c *Cleaner) Run(ctx context.Context) error {
 			continue
 		}
 
-		newStoredVersions, matchedStoredVersion := removeMatchingStoredVersions(crd.Status.StoredVersions, deprecatedVersionRegexp)
+		newStoredVersions, deprecatedVersion := removeMatchingStoredVersions(crd.Status.StoredVersions, deprecatedVersionRegexp)
 
 		// If there is no new version found other than the matched version, we short circuit here, as there is no updated version found in the CRDs
 		if len(newStoredVersions) <= 0 {
-			return errors.New(fmt.Sprintf("it doesn't look like your version of ASO is one that supports deprecating version %q. Have you upgraded ASO yet?", matchedStoredVersion))
+			return errors.New(fmt.Sprintf("it doesn't look like your version of ASO is one that supports deprecating version %q. Have you upgraded ASO yet?", deprecatedVersion))
 		}
 
+		activeVersion := newStoredVersions[0]
+
 		// If we found an updated slice, which implies, we have found a version to deprecate. Then only continue with the cleaning process
 		if len(newStoredVersions) != len(crd.Status.StoredVersions) {
 			klog.Infof("Starting cleanup for %q", crd.Name)
-			objectsToMigrate, err := c.getObjectsForMigration(ctx, crd, matchedStoredVersion)
+			objectsToMigrate, err := c.getObjectsForMigration(ctx, crd, activeVersion)
```

Comment:
Why would we get objects for migration with activeVersion?

Reply:
We're supposed to get the objects for migration, which are under deprecatedVersion, and then issue a PUT onto them?

Reply:
So the issue is that we want `asoctl` to successfully be able to remove deprecated API versions even if the user is running a newer `asoctl` and they're on an older cluster. 

For example, when when built with `v2.0.0`, or even `v2.1.0`.

The code we had before though, was trying to use the deprecated api version to perform the `c.client.List`. The issue is, that deprecated version isn't in the `scheme` of `v2.0.0` (because it's been removed/deprecated), so that list will fail.

TL;DR: When migrating CRs, we need to use an _active_ version to migrate them, not a deprecated version.